### PR TITLE
Merge fix for node stall ready search

### LINKED ISSUE #28: Node Stall Ready check does not perform search in stall queue as intended
When sending a message via a node, first bidib determines if the node is ready to send.    
This involves, among other things, checking whether the node is *stalled*, or, whether a supernode of the node is *stalled*. The stall check is performed in a function [here](https://github.com/uniba-swt/libbidib/blob/master/src/transmission/bidib_transmission_node_states.c#L108).

If the node or a supernode is stalled, the linked function returns *false*. A node is stalled if the member "stall" is true and the address_stack (input param to linked fct) is NOT contained in the member *stall_affected_nodes_queue*.   
Thus, the linked fct attempts to find the addr_stack in the queue *stall_affected_nodes_queue* by using g_queue_find, see line 114. As the addr_stack is an array of 4 uint8_t's, a custom comparison function is used, as the comparison shall be element-wise, not a pointer comparison. This comparator is called `bidib_node_stall_queue_entry_equals` (see line 99ff.).    
The call to `g_queue_find` looks as follows: `!g_queue_find(state->stall_affected_nodes_queue, bidib_node_stall_queue_entry_equals)`.    
Note that the element to find is not actually given here! This code just always returns true, the comparator is never called. `g_queue_find` is supposed to be called with the queue and the element to find ([see here](https://docs.gtk.org/glib/method.Queue.find.html)). To use a custom comparator, we need to use `g_queue_find_custom` ([see here](https://docs.gtk.org/glib/method.Queue.find_custom.html)). It shall be called like this: `!g_queue_find_custom(state->stall_affected_nodes_queue, addr_stack, (GCompareFunc)bidib_node_stall_queue_entry_equals)`.

The unit tests related to this part (bidib_send_tests) pass as the "stall" member is apparently sufficient to indicate whether the node is stalled or not. However, there's probably a reason for the existence of the stall_affected_nodes_queue, so this comparison call should be fixed.

### PR DESCRIPTION
**Pull request by BLuedtke**
Closes #28 

Draft because tests still have to be run on how this affects overall operation. More info available in issue #28

### CODE REVIEW

**Comment by BLuedtke** [comment]
Looking at it again, it is now clear to me why this worked before the fix.

The point of searching the `stall_affected_nodes_queue` was to see if a node was already in there - if not, add it. Later, in `bidib_node_update_stall`, when the node is not stalled anymore, all entries in the `stall_affected_nodes_queue` are popped and for each entry, we try to send any queued messages.

With the old code where the search function for the `stall_affected_nodes_queue` did not work, it would always return false -> so the node might be added to the stall affected nodes queue multiple times. But that just means that when the node is not stalled anymore, and it has multiple identical entries in the `stall_affected_nodes_queue`, it just tries to send queued messages multiple times, which is no problem.

So the fix only prevents the same node from being added to the stall affected nodes queue multiple times, nothing else.

File: src/transmission/bidib_transmission_node_states.c [hunk 1]
```diff
@@ -105,20 +105,36 @@ static int bidib_node_stall_queue_entry_equals(const t_bidib_stall_queue_entry *
...
-		    !g_queue_find(state->stall_affected_nodes_queue,
-		                  bidib_node_stall_queue_entry_equals)) {
-			t_bidib_stall_queue_entry *stall_entry = malloc(
-					sizeof(t_bidib_stall_queue_entry));
-			memcpy(stall_entry->addr, addr_stack, 4);
-			g_queue_push_tail(state->stall_affected_nodes_queue, stall_entry);
+		// Is the node at the address of current addr_cpy stalled?
+		if (state != NULL && state->stall) {
+			// Node at addr_cpy is stalled -> search its stall_affected_nodes_queue to see
+			// if (the node at) addr_stack is contained in it.
```
Comment (eyip002) [inline src/transmission/bidib_transmission_node_states.c]: `if the queue contains the (node at) addr_stack.`

**Review by eyip002 — DISMISSED** [review]
Changes look good. Minor suggestion for a comment.

**Comment by BLuedtke** [comment]
Huh? Not sure why it says I dismissed the review. Let me see if I can revert that.

Edit: Can't see where I can revert it. Makes no sense, not sure how this happened. GitHub is acting weird wrt. merge requests recently, sometimes it links completely unrelated issues automatically.

**Pull request merged**